src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  let newState = state;
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. You should make copy using `Object.assign` or `spread` operator. The variable `newState` should be a copy of `state` from the beginning to prevent mutating the original `state` object.